homeassistant/helpers/entity_registry.py

```diff
@@ -1832,10 +1836,18 @@ def _write_unavailable_states(_: Event) -> None:
         """Make sure state machine contains entry for each registered entity."""
         existing = set(hass.states.async_entity_ids())
 
+        entity_sources = entity.entity_sources(hass)
         for entry in registry.entities.values():
             if entry.entity_id in existing or entry.disabled:
                 continue
 
+            entity_info: entity.EntityInfo = {
+                "domain": entry.platform,
+            }
+            if entry.config_entry_id:
+                entity_info["config_entry"] = entry.config_entry_id
+
+            entity_sources[entry.entity_id] = entity_info
```

Comment:
since `entity.async_internal_added_to_hass` already adds the entity to `entity_sources` why is this needed here too?
assuming it is, couldn't we somehow move it to the Entity class, and avoid a circular dependency?

Reply:
This function (`_write_unavailable_states`) is called to create `unavailable` states for all entities known to the entity registry. There are no backing entity objects.

The point of this PR is to ensure we create entries in `entity_sources` for these `unavailable` states.

Reply:
It _is_ stored in `hass.data`; `homeassistant.helpers.entity.entity_sources` returns `hass.data["entity_info]"`